controllers/etcd_controller.go

```diff
@@ -262,7 +318,15 @@ func (r *EtcdReconciler) delete(ctx context.Context, etcd *druidv1alpha1.Etcd) (
 		}, err
 	}
 
-	if err := r.removeFinalizersToDependantSecrets(ctx, etcd); err != nil {
+	if etcd.Spec.VolumeRetentionPolicy == nil || *etcd.Spec.VolumeRetentionPolicy == druidv1alpha1.VolumeRetentionPolicyDelete {
```

Comment:
Is [this](https://github.com/gardener/etcd-druid/pull/59/files#diff-8e5c816f8b2f1adeef36a64d81b3e0f6R650) not required here?

Reply:
We do have it. Druid will take a full snapshot before removing the statefulset, by calling `takeFullSnapshot()`. As per your suggestion, I've replaced it with `prepareEtcdForScaleDown()` for efficient code reuse.